common/determined_common/experimental/checkpoint/_torch.py

```diff
@@ -10,7 +10,8 @@
 def load_model(
     ckpt_dir: pathlib.Path, metadata: Dict[str, Any], **kwargs: Any
 ) -> Union[PyTorchTrial, torch.nn.Module]:
-    checkpoint = torch.load(ckpt_dir.joinpath("state_dict.pth"), **kwargs)  # type: ignore
+    with open(ckpt_dir.joinpath("state_dict.pth"), "rb") as opened_file:
```

Comment:
if you stick to this strategy, probably calling `ckpt_dir.joinpath("state_dict.pth").open(mode="rb")` is preferable

Reply:
Huh, str(...) actually works. I assumed everything path-like would be treated the same. Going with that, and in _pytorch_trial.py too. What's the consequential difference between the 2 opening styles, btw?

Reply:
> What's the consequential difference between the 2 opening styles, btw?

`pathlib.Path` doesn't do anything that `os.path`, `os`, and the built-in `open`/`close` don't do.  It's really just syntactic sugar, presenting an object-oriented interface to file paths and various system calls.  While you totally can call `open()` on a pathlib object, it's weird to use `pathlib` and not use `pathlib.Path.open()` instead, like you should really pick one strategy or the other.

But I don't think there's any material difference at all.